sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java

```diff
@@ -1,20 +1,742 @@
+          || Build.MANUFACTURER.contains("Genymotion")
+          || Build.PRODUCT.contains("sdk_google")
+          || Build.PRODUCT.contains("google_sdk")
+          || Build.PRODUCT.contains("sdk")
+          || Build.PRODUCT.contains("sdk_x86")
+          || Build.PRODUCT.contains("vbox86p")
+          || Build.PRODUCT.contains("emulator")
+          || Build.PRODUCT.contains("simulator");
+    } catch (Exception e) {
+      log(SentryLevel.ERROR, "Error checking whether application is running in an emulator.", e);
+      return null;
+    }
+  }
+
+  /**
+   * Get the total amount of internal storage, in bytes.
+   *
+   * @return the total amount of internal storage, in bytes
+   */
+  private Long getTotalInternalStorage(StatFs stat) {
+    try {
+      long blockSize = getBlockSizeLong(stat);
+      long totalBlocks = getBlockCountLong(stat);
+      return totalBlocks * blockSize;
+    } catch (Exception e) {
+      log(SentryLevel.ERROR, "Error getting total internal storage amount.", e);
+      return null;
+    }
+  }
+
+  @SuppressWarnings("ObsoleteSdkInt")
+  private long getBlockSizeLong(StatFs stat) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
+      return stat.getBlockSizeLong();
+    }
+    return getBlockSizeDep(stat);
+  }
+
+  @SuppressWarnings("deprecation")
+  private int getBlockSizeDep(StatFs stat) {
+    return stat.getBlockSize();
+  }
+
+  @SuppressWarnings("ObsoleteSdkInt")
+  private long getBlockCountLong(StatFs stat) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
+      return stat.getBlockCountLong();
+    }
+    return getBlockCountDep(stat);
+  }
+
+  @SuppressWarnings("deprecation")
+  private int getBlockCountDep(StatFs stat) {
+    return stat.getBlockCount();
+  }
+
+  @SuppressWarnings("ObsoleteSdkInt")
+  private long getAvailableBlocksLong(StatFs stat) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
+      return stat.getAvailableBlocksLong();
+    }
+    return getAvailableBlocksDep(stat);
+  }
+
+  @SuppressWarnings("deprecation")
+  private int getAvailableBlocksDep(StatFs stat) {
+    return stat.getAvailableBlocks();
+  }
+
+  /**
+   * Get the unused amount of internal storage, in bytes.
+   *
+   * @return the unused amount of internal storage, in bytes
+   */
+  private Long getUnusedInternalStorage(StatFs stat) {
+    try {
+      long blockSize = getBlockSizeLong(stat);
+      long availableBlocks = getAvailableBlocksLong(stat);
+      return availableBlocks * blockSize;
+    } catch (Exception e) {
+      log(SentryLevel.ERROR, "Error getting unused internal storage amount.", e);
+      return null;
+    }
+  }
+
+  private StatFs getInternalStorageStat() {
+    File path = Environment.getDataDirectory(); // /data
+    return new StatFs(path.getPath());
+  }
+
+  private StatFs getExternalStorageStat() {
+    // might need READ_EXTERNAL_STORAGE permission
```

Comment:
Can we test for the permission and only call it if the app has it?
IMHO not worth requesting permission for the library so that we can report this.

Reply:
yeah, I'll do it together with https://github.com/getsentry/sentry-android/pull/35#discussion_r335923486 if necessary.